report/src/index.ts

```diff
@@ -0,0 +1,361 @@
+                        color:${config.blueColor};
+                    " class="fa ${row.given_attribute_type.slice(0, 6) === "string" ? "fa-font" : "fa-hashtag"}"></i>
+                    <b>${row.attribute}</b>
+                    <span style="float:right;">${row.given_attribute_type}</span>
+                </span>
+                <span style="padding:12px 2px;display:inline-block;font-family: Verdana; color: rgb(51, 51, 51); font-weight: bold; font-size: 14px;">
+                    Optimal:
+                </span>
+                <span style="
+                    color:${config.primaryColor};
+                    padding:4px 8px;
+                    display:inline-block;
+                    font-size:${config.secondaryFontSize}px;
+                    margin-top:4px;
+                    border:1px solid ${config.secondaryColor};
+                    border-radius:4px;
+                    background-color: ${config.offwhiteColor};
+                    float:right;
+                ">
+                    <i style="
+                        font-size:${config.secondaryFontSize}px;
+                        color:${config.blueColor};
+                    " class="fa ${row.best_attribute_type.slice(0, 6) === "string" ? "fa-font" : "fa-hashtag"}"></i>
+                    ${row.best_attribute_type}
+                </span>
+                `)
+                .overflowX("hidden")
+                .overflowY("hidden")
+                ;
+            let fillRate = row.fill_rate === 100 || row.fill_rate === 0 ? row.fill_rate : row.fill_rate.toFixed(1);
+            const w2 = new StyledTable()
+                .data([
+                    ["Cardinality", row.cardinality, "(~" + (row.cardinality / row.fill_count * 100).toFixed(0) + "%)"],
+                    ["Filled", row.fill_count, fillRate <= config.fillRateRedThreshold ? `(<b style="color:${config.redColor}">` + fillRate + "%</b>)" : "(" + fillRate + "%)"],
+                ])
+                .tbodyColumnStyles([
+                    { "font-weight": "bold", "font-size": config.secondaryFontSize + "px", "width": "1%" },
+                    { "font-weight": "normal", "font-size": config.secondaryFontSize + "px", "text-align": "right", "width": "auto" },
+                    { "font-weight": "normal", "font-size": config.secondaryFontSize + "px", "text-align": "left", "width": "1%" },
+                ])
+                ;
+            const grid = new Grid()
+                .gutter(0)
+                .surfaceShadow(false)
+                .surfacePadding("0")
+                .surfaceBorderWidth(0)
+                .setContent(0, 0, w1)
+                .setContent(1, 0, w2)
+                ;
+            return grid;
+        }
+        function getBreakdownWidget(row) {
+            if (row.cardinality_breakdown.Row.length > 0) {
+                return new BreakdownTable()
+                    .columns(["Cardinality Breakdown", ""])
```

Comment:
Should probably change this to "Cardinality" to prevent this column's content from getting cut off at lower resolutions.